src/test/java/org/wso2/carbon/connector/integration/test/StackExchangeConnectorIntegrationTest.java

```diff
@@ -0,0 +1,699 @@
+    /* ============================================= Question routes ============================================= */
+
+    private StackExchangeItems getStackExchangeQuestionItems() throws Exception {
+
+        StackExchangeUrl questionUrl =
+                new StackExchangeUrl.Builder(apiVersion, "/search/advanced")
+                        .queryParam("site", site)
+                        .queryParam("accepted", "False")
+                        .queryParam("answers", "1").build();
+        return getStackExchangeItems(questionUrl);
+    }
+
+    /* ============================================= Answer routes ============================================= */
+
+    private StackExchangeItems getStackExchangeAnswerItems(int questionId, String site) throws Exception {
+
+        StackExchangeUrl answerUrl =
+                new StackExchangeUrl.Builder(apiVersion, "/questions/" + questionId + "/answers")
+                        .queryParam("site", site).build();
+        return getStackExchangeItems(answerUrl);
+    }
+
+    private StackExchangeItems getStackExchangeAnswerItems(int questionId) throws Exception {
+
+        StackExchangeUrl answerUrl =
+                new StackExchangeUrl.Builder(apiVersion, "/questions/" + questionId + "/answers")
+                        .queryParam("site", site).build();
+        return getStackExchangeItems(answerUrl);
+    }
+
+    /* ============================================= Privilege routes ============================================= */
+
+    private StackExchangeItems getStackExchangePrivilegeItems(String accessToken) throws Exception {
+
+        StackExchangeUrl privilegeUrl =
+                new StackExchangeUrl.Builder(apiVersion, "/me/privileges/")
+                        .queryParam("site", site)
+                        .queryParam("key", key)
+                        .queryParam("access_token", accessToken).build();
+        return getStackExchangeItems(privilegeUrl);
+    }
+
+    private <T> void setListLikePropertyInSystem(String propertyName, T[] array) {
+
+        stackExchangeProperties.setProperty(propertyName, getTArrayAsSemicolonDelimitedString(array));
+    }
+
+    private <T> void setListLikePropertyInConnector(String propertyName, T[] array) {
+
+        connectorProperties.setProperty(propertyName, getTArrayAsSemicolonDelimitedString(array));
+    }
+
+    private <T> String getTArrayAsSemicolonDelimitedString(T[] array) {
+
+        StringBuilder builder = new StringBuilder();
+        for (T a : array) {
+            builder.append(a).append(";");
+        }
+        if (builder.length() > 0) {
+            builder.setLength(builder.length() - 1);
+        }
+        return builder.toString();
+    }
+
+    /* ======================================= getUsersByIds ======================================= */
+
+    @StackExchange(skipPrivilegeCheck = true)
+    @Test(groups = {"wso2.ei"})
+    public void testUsersByIdsWithInvalid() throws IOException, JSONException {
```

Comment:
Add method level comments for all the test methods

Reply:
Fix with 503a067